src/starterbot/BaseManager.cpp

```diff
@@ -0,0 +1,156 @@
+#include "BaseManager.h"
+
+BaseManager::BaseManager(UnitManager& unitManager) :
+    m_unitManager(unitManager) {
+}
+
+Base& BaseManager::getClosestBase(bw::Unit& unit) {
+    Base* closestBase;
+    int shortestDistance = INT_MAX;
+
+    // getDistance uses Brood War's built-in pathfinding, so this will find the
+    // base that's closest while accounting for terrain.
+    for (Base& base : m_bases) {
+        int distance = unit->getDistance(base.centroid);
+
+        if (distance < shortestDistance) {
+            shortestDistance = distance;
+            closestBase = &base;
+        }
+    }
+
+    return *closestBase;
+}
+
+// Perform basic map analysis by finding all resources on the map, clustering them, and 
+// creating Bases at them.
+void BaseManager::onStart() {
+
+    std::vector<bw::Unitset> resourceClusters;
```

Comment:
(As mentioned in Discord) We can probably use the clustering algorithm from #22 here.